### PR TITLE
OCPBUGS-22350: address CVE-2023-44487

### PR DESCRIPTION
**Pull request by alebedev87**
What was done to address CVE-2023-44487 ([Operator Remediation Guidance followed](https://docs.google.com/document/d/1F99glzhX2i2Ppe5Qq3M-Po_jSSUiVssqFyWRhVm1X3o)):
- bump controller-runtime v0.15.3
  - bump of `x/net` and kube api were done along the way
- bump the operand image to [the latest version](https://quay.io/repository/external-dns-operator/external-dns/manifest/sha256:6fa3afebbc99926163d4c827a638473e106e6589e9f3d3d3accd3e426693f04f) (`0e095a3`)
- disable HTTP2 for kube-rbac-proxy container
- disable HTTP2 for webhook server

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func main() {
 	flag.StringVar(&opCfg.HealthProbeBindAddress, "health-probe-bind-addr", operatorconfig.DefaultHealthProbeAddr, "The address the health endpoint binds to.")
 	flag.IntVar(&opCfg.RequeuePeriodSeconds, "requeue-period", operatorconfig.DefaultRequeuePeriodSeconds, "Requeue period for a failed reconciliation (in seconds).")
 	flag.BoolVar(&opCfg.EnableLeaderElection, "leader-elect", operatorconfig.DefaultEnableLeaderElection, "Enable leader election for controller manager to ensure there is only one active controller manager.")
+	flag.BoolVar(&opCfg.WebhookDisableHTTP2, "webhook-disable-http2", false, "Disable HTTP2 for the webhook server.")
```
Comment (Miciah) [inline main.go:52]: It's usually written "HTTP/2", but this isn't worth worrying about if it's the only issue that comes up during review.

**Comment by Miciah** [comment]
Thanks!
/approve
/lgtm
/hold
for @gcs278.

File: main.go (hunk 1 above)
Comment (alebedev87) [reply main.go]: Done.

**Comment by gcs278** [comment]
I have some questions about how the ExternalDNS bump works, but I'll follow up with you offline.
/unhold
/lgtm

**Pull request merged**